site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,995 @@
+                "specified": "OVERFLOW_ERROR"
+              }
+            },
+            {
+              "value": {
+                "scalarFunction": {
+                  "functionReference": 4,
+                  "outputType": {
+                    "decimal": {
+                      "precision": 38,
+                      "scale": 2,
+                      "nullability": "NULLABILITY_NULLABLE"
+                    }
+                  },
+                  "arguments": [
+                    {
+                      "enum": {
+                        "specified": "OVERFLOW_ERROR"
+                      }
+                    },
+                    {
+                      "value": {
+                        "cast": {
+                          "type": {
+                            "decimal": {
+                              "precision": 10,
+                              "scale": 2,
+                              "nullability": "NULLABILITY_REQUIRED"
+                            }
+                          },
+                          "input": {
+                            "selection": {
+                              "directReference": {
+                                "structField": {
+                                  "field": 1
+                                }
+                              },
+                              "rootReference": {}
+                            }
+                          }
+                        }
+                      }
+                    },
+                    {
+                      "value": {
+                        "selection": {
+                          "directReference": {
+                            "structField": {
+                              "field": 3
+                            }
+                          },
+                          "rootReference": {}
+                        }
+                      }
+                    }
+                  ]
+                }
+              }
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+## Field indices
+
+So far, we have glossed over the field indices. Now that we've built up each
+of the relations, it will be a bit easier to explain them.
+
+Throughout the plan, data always has some implicit schema, which is modified
+by each relation. Often, the schema can change *within* a relation--we'll discuss
+an example in the next section. Each relation has it's own rules in how schemas
+are modified. For the purposes of our query, the relevant rules are:
+
+* For Read relations, their output schema is the schema of the table.
+* For Filter relations, the output schema is the same as in the input schema.
+* For Joins relations, the input schema is the concatenation of the left and then the
+  right schemas. The output schema is the same.
+* For Aggregate relations, the output schema is the group by fields followed by
+  the measures.
+
+<!-- TODO: do we document these rules somewhere? -->
```

Comment:
In the docs / spec this is referred to as the "direct output order"

![image](https://user-images.githubusercontent.com/1696093/206299475-17a02cc2-3f57-47e9-9e3d-bf027c62da6b.png)

Reply:
Also seems like "emit order" is used. I've added mention of those two phrases to the doc so readers will recognize them.